src/parser/style-context.ts

```diff
@@ -121,6 +141,24 @@ export function styleNodeRange(
   ];
 }
 
+/**
+ * Extracts a node location (like that of any ESLint node) from a parsed svelte selector node.
+ */
+export function styleSelectorNodeLoc(
+  node: SelectorNode,
+): Partial<SourceLocation> {
+  return {
+    start:
+      node.source?.start !== undefined
+        ? {
+            line: node.source.start.line,
+            column: node.source.start.column - 1,
+          }
+        : undefined,
+    end: node.source?.end,
+  };
+}
```

Comment:
Why don't you do this in `fixSelectorNodeLocation`? I thought the `loc` information could always be handled in ESLint style because this program runs on ESLint.

Reply:
Yeah, I'm following the same approach we agreed on in https://github.com/sveltejs/svelte-eslint-parser/pull/340#discussion_r1208460662 - the TL;DR is that PostCSS AST locations are different than ESLint AST locations - one of them starts counting columns from 0 and other from 1, one of them has the end point at the last character while the other has it point at the first character after the end... So we agreed we would recalculate the positions to be relative to the whole file instead of being relative to the selector, but we would keep the PostCSS semantics in the PostCSS AST. Only when converting from PostCSS AST to an ESLint `SourceLocation` would we convert this to the ESLint way of doing things.

Reply:
Of course this is debatable (there is no universally "right" solution IMO), but I'd strongly suggest doing the selector parsing in the same way as the style node parsing. So if we were to change this one, we'd need to change the other one too.

Reply:
Thank you for explanation! OK. So let's merge this PR now!